search

anasty223 opened the pull request:
1.создан файл Search с кодом-заготовкой под рендер карточек
2. создана папка tamplate в ней файл film-cards.hbs -заготовка
3.в файле fetch-api.js фетч на поиск фильмов

anasty223 committed:
search

1.создан файл Search с заготовкой под рендер карточек
2. создана папка tamplate в ней файл film-cards.hbs -заготовка
3.в файле fetch-api.js фетч на поиск фильмов

VladimirBludov committed:
Delete package-lock.json

VladimirBludov commented on src/js/Search.js:
```diff
@@ -0,0 +1,35 @@
+const { serchKeyword, gallery} = refs;
+
+const newsApiService = new NewsApiService();
+serchKeyword.addEventListener('submit', onSearch);
```
событие сабмит отлавливай на форме, а не на кнопке

VladimirBludov commented on src/js/Search.js:
```diff
@@ -0,0 +1,35 @@
+function onSearch(e) {
+  e.preventDefault();
+
+  let searchQuery =e.target[0].value;  //e.currentTarget.elements.searchQuery.value.trim();
```
на форме есть псевдомасив, elements. после того как начнешь отлавливать на форме, значение импута бери оттуда

VladimirBludov commented on src/js/Search.js:
```diff
@@ -0,0 +1,35 @@
+import NewsApiService from './api/fetch-api.js'
```
NewsApiService почему News? мы вроде фильмы ищем?))

VladimirBludov commented on src/js/components/film-description-modal.js:
```diff
@@ -1 +0,0 @@
-
```
верни на место)

VladimirBludov commented on src/js/components/refs.js:
```diff
@@ -10,6 +10,6 @@ export const refs = {
-
+   serchKeyword: document.querySelector('#js-search'),
+ gallery: document.querySelector('.gallery'),
 };
```
добавь пустую строку в конце

VladimirBludov commented on package.json:
```diff
@@ -24,5 +24,10 @@
     "sass": "^1.32.8"
+  },
+  "dependencies": {
+    "axios": "^0.24.0",
```
зачем тебе axios, если ты его не используешь?

VladimirBludov commented on src/index.js:
```diff
@@ -1,3 +1,5 @@
 
 import './js/components/togglePages'
+
+import './js/Search';
```
добавь пустую строку в конце

VladimirBludov commented on src/js/Search.js:
```diff
@@ -0,0 +1,35 @@
+import NewsApiService from './api/fetch-api.js'
+import { refs } from './components/refs.js';
```
файл refs.js перенеси в utils, не забудь потом изменить у себя путь

VladimirBludov commented on src/partials/sections/films-container.html:
```diff
@@ -1,42 +1,7 @@
   <h2 class="visually-hidden">Films</h2>
   <!-- <include src="./partials/utils/spinner.html"></include> -->
-  <div class="films__container">
-    <div class="films__card">
```
зачем ты удалила настроеный контейнер для фильмов <div class="films__container">? используй его а не секцию (она не настроена и так неправильно), дополнительный класс gallery незачем, используй в рефах films__container

VladimirBludov commented on src/partials/sections/header.html:
```diff
@@ -1,5 +1,5 @@
 <header class="header" id="home">
-  <div class="container">
+  <div class="container gallery">
     <div class="header__wrapper">
```
зачем тебе в хедере gallery, по твоему коду, фильмы еще и сюда будут рендериться)

VladimirBludov reviewed (changes requested):
src/tamplates/film-cards.hbs не нужен, Женя уже почти доделал карточку, убери

anasty223 committed:
bag-fix

1.удалила ненужную tamplates
2.переименовала клас
3.добавила пустые строки
4.удалила ненужный axios
5.перенесла refs в utils
6.добавила контейнер
7.использовала в рефах films__container вместо gallery

VladimirBludov committed:
Merge branch 'main' into search

VladimirBludov committed:
fix-project-settings

VladimirBludov committed:
Merge branch 'main' into search

VladimirBludov committed:
fix
